src/python/structurefinder/pyberny/__init__.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2023 NWChemEx-Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pluginplay as pp
+from friendzone.nwx2qcelemental import chemical_system_conversions
+from simde import EnergyNuclearGradientStdVectorD, TotalEnergy
+from berny import Berny, geomlib, optimize
+import chemist
+import qcelemental as qcel
+
+
+class GeomoptViaPyberny(pp.ModuleBase):
+
+    def __init__(self):
+        pp.ModuleBase.__init__(self)
+        self.satisfies_property_type(EnergyNuclearGradientStdVectorD())
```

Comment:
```suggestion
```
I assume you added this so you could just take a PointSet? Once the geometry changes, the PointSet the user gives you won't be the one you want to take the derivative with respect to, so that won't work. You still would need to get the PointSet from the geometry.

Reply:
I added this because I thought in order to utilize the `NWCHem : SCF Gradient` module it would need to be able to satisfy that property type, similar to how `TotalEnergy()` is used when using `NWChem : SCF` when calculating the energy. The way I've been thinking about it is if `mm.run_as()` is called, I am specifying the property type from SimDE that I am satisfying (and thus the type of calculation I am doing), I am then specifying the module I want to run that incorporates that property type, and then I give the input.

Reply:
Gotcha. So you're logic is along the right line of thought, but is confusing the optimizer module with the modules it calls. To better understand this consider two functions:
```.py
def foo(x):
    return x + 1

def bar(x):
    return str(foo(x))
```

Assume that in both cases `x` is an integer. `foo` then takes an integer and returns a different one, whereas `bar` takes an integer and then returns a string. The [function signature](https://stackoverflow.com/questions/72788932/what-is-a-function-signature) of `foo` is then written `foo(int)->int` and that of `bar` is `bar(int)->str`. It would be incorrect to say that `bar` has the signature `bar(int)->int` even though it calls `foo`.

Modules are essentially functions. So your `GeomoptViaPyberny` module is analogous to `bar` in the above example in the sense that it is a module that calls another module, just like `bar` calls another function `foo`. Property types are essentially a way to represent a module's signature programmatically. Right now the signature for `GeomoptViaPyberny` is `double GeomoptViaPyberny(ChemicalSystem)` which is codified by the `TotalEnergy` property type. If `GeomoptViaPyberny` satisfied the  `EnergyNuclearGradientStdVectorD` property type it would have the signature `[double] GeomoptPyberny(ChemicalSystem, PointSetD)`, which is not the case; it's the module called by `GeomoptViaPyberny` that has the `EnergyNuclearGraidentStdVectorD` signature.

Does that help?

Reply:
This helps out a lot! I was able to get just the `mm.run_as(simde.EnergyNuclearGradientStdVectorD(), "NWChem : SCF Gradient:, <Chemical System>, chemist.PointSetD())` to work but couldn't with PyBerny, which now makes sense as to why. The link to the explanation of the function signature helps as well.